trl/trainer/dpo_trainer.py

```diff
@@ -723,9 +723,18 @@ def build_tokenized_answer(self, prompt, answer, images=None):
         if self.is_vision_model:
             if answer.count("<image>") > 0:
                 raise NotImplementedError("Answer contains <image> token, which is not supported yet.")
-            full_tokenized = self.processor(prompt + answer, images=images, add_special_tokens=False)
+            if "add_special_tokens" in inspect.signature(self.processor).parameters:
+                processor_kwargs = {"add_special_tokens": False}
+            else:
+                processor_kwargs = {}
+            full_tokenized = self.processor(prompt + answer, images=images, **processor_kwargs)
```

Comment:
`add_special_token` is not an argument in all processors. So the only way is to check the signature.